test/lib/rules/no-unknown-wire-adapters.js

```diff
@@ -0,0 +1,202 @@
+            code: `class Test {
+                @foo
+                foo;
+
+                @wire
+                wiredPropA;
+
+                @wire()
+                wiredPropB;
+
+                @wire({})
+                wiredPropC;
+            }`,
+            options: DEFAULT_OPTIONS,
+        },
+        {
+            code: `import { wire } from 'lwc';
+            import { getFoo } from 'adapter';
+
+            class Test {
+                @wire(getFoo)
+                wiredProp;
+            }`,
+            options: [
+                {
+                    adapters: [{ module: 'adapter', identifier: 'getFoo' }],
+                },
+            ],
+        },
+        {
+            code: `import { wire } from 'lwc';
+            import { getFoo as getBar } from 'adapter';
+
+            class Test {
+                @wire(getBar)
+                wiredProp;
+            }`,
+            options: [
+                {
+                    adapters: [{ module: 'adapter', identifier: 'getFoo' }],
+                },
+            ],
+        },
+        {
+            code: `import { wire } from 'lwc';
+            import getFoo from 'adapter';
+
+            class Test {
+                @wire(getFoo)
+                wiredProp;
+            }`,
+            options: [
+                {
+                    adapters: [{ module: 'adapter', identifier: 'default' }],
+                },
+            ],
+        },
+        {
+            code: `import { wire } from 'lwc';
+            import getFoo, { getBar } from 'adapter';
+
+            class Test {
+                @wire(getFoo)
+                wiredFoo;
+
+                @wire(getBar)
+                wiredBar;
+            }`,
+            options: [
+                {
+                    adapters: [
+                        { module: 'adapter', identifier: 'default' },
+                        { module: 'adapter', identifier: 'getBar' },
+                    ],
+                },
+            ],
+        },
+        {
+            code: `import { wire } from 'lwc';
+            import { getFoo } from 'adapterFoo';
+            import { getBar } from 'adapterBar';
+
+            class Test {
+                @wire(getFoo)
+                wiredFoo;
+
+                @wire(getBar)
+                wiredBar;
+            }`,
+            options: [
+                {
+                    adapters: [
+                        { module: 'adapterFoo', identifier: 'getFoo' },
+                        { module: 'adapterBar', identifier: 'getBar' },
+                    ],
+                },
+            ],
+        },
+    ],
+    invalid: [
+        {
+            code: `import { wire } from 'lwc';
+
+            class Test {
+                @wire(getFoo) wiredProp;
+            }`,
+            options: DEFAULT_OPTIONS,
+            errors: [
+                {
+                    message: '"getFoo" is not a known adapter.',
```

Comment:
What do you think about making this more obvious? Eg `@wire(getFoo) is not a known wire adapter.` or `”getFoo” in @wire(getFoo) is not a known wire adapter.`

Reply:
All the standard eslint rules don't provide information about the warning/error location hint in the message because the location is already attached to the warning/error when passing the associated `node` to `context.report()`.

IMO, since the derived location from `node` provide an exact start and end location in the original source code, adding another location hint in the error message doesn't add any value.